src/cmap/connection_pool.ts

```diff
@@ -437,7 +525,31 @@ function createConnection(pool: ConnectionPool, callback?: Callback<Connection>)
       connection.on(event, (e: any) => pool.emit(event, e));
     }
 
-    pool.emit(ConnectionPool.CONNECTION_POOL_CREATED, new ConnectionCreatedEvent(pool, connection));
+    pool.emit(ConnectionPool.CONNECTION_CREATED, new ConnectionCreatedEvent(pool, connection));
+
+    if (pool.loadBalanced) {
+      const onPin = (pinType: string) => {
+        pool[kMetrics].markPinned(pinType);
+      };
+      const onUnpin = (pinType: string) => {
+        pool[kMetrics].markUnpinned(pinType);
+      };
+      connection.on('pinned', onPin);
+      connection.on('unpin', onUnpin);
+
+      const serviceId = connection.serviceId;
+      if (serviceId) {
+        let generation;
```

Comment:
`let generation: number;` -> then you won't have to cast it on line 550.

Reply:
Actually that didn't work in this case and gives the same error. Typescript doesn't seem to be smart enough to know it will never be undefined even with the check if it doesn't exist in the code. Maybe it's not smart enough for Maps?

Reply:
Ah yea because `.get` can return undefined there isn't a great way to narrow it is my understanding, fine to leave as is.